### ISSUE TITLE
Cant log boosted sltr query

### ISSUE THREAD

**Issue by softwaredoug** [o19s/elasticsearch-learning-to-rank#70]
To address #69, (so I can log features for a set of ids), I added an sltr query with a boost of 0. Specifically:

```
GET tmdb/_search
{
    "explain": true, 
    "query": {
        "bool": {
            "should": [                
                {"sltr": {
                    "boost": 0,
                    "_name": "logged_featureset",
                    "featureset": "movie_features",
                    "params": {
                        "keywords": "rambo"
                    }
                }},
                {"match": {
                   "overview": 
                   {
                       "query": "rambo"
                   }
                }}
                ]
            }
    },
    "ext": {
        "ltr_log": {
            "log_specs": {
                "name": "log_entry1",
                "named_query": "logged_featureset"
            }
        }
    }
}
```

This gives the following error:

```
{
   "error": {
      "root_cause": [
         {
            "type": "illegal_argument_exception",
            "reason": "Query named [logged_featureset] must be a [sltr] query [BoostQuery] found"
         }
      ],
      "type": "search_phase_execution_exception",
      "reason": "all shards failed",
      "phase": "query",
      "grouped": true,
      "failed_shards": [
         {
            "shard": 0,
            "index": "tmdb",
            "node": "A1yBd5opScyEeVTitmkyCA",
            "reason": {
               "type": "illegal_argument_exception",
               "reason": "Query named [logged_featureset] must be a [sltr] query [BoostQuery] found"
            }
         }
      ]
   },
   "status": 400
}
```

**Comment by nomoa** [comment]
indeed the logging code is not smart enough to inspect the query behing the BoostQuery, the fix should be fairly easy, at https://github.com/o19s/elasticsearch-learning-to-rank/blob/74c3272c293bebd901c3bd255ab776f20a147a6d/src/main/java/com/o19s/es/ltr/logging/LoggingFetchSubPhase.java#L132
we should make the code accepts BoostQuery and inspect the boosted query.

**Comment by softwaredoug** [comment]
Fixed in #73
